upsertGitHubTag/deployment/index.js

```diff
@@ -252,9 +254,41 @@ function processEvent(event, callback) {
         githubEventType +
         " from GitHub.",
     });
+    return;
+  }
+  // If bucket name is not null (had to put this for the integration test)
+  if (process.env.BUCKET_NAME) {
```

Comment:
I make this part of the `logPayloadToS3` method, `logPayloadtoS3(body, bucketPath, deliveryId)`. 

> How strongly do we feel about moving my s3 code to be before the callback? I avoided putting my code to s3 before the callback to avoid too many if-blocks because we want to avoid submitting the event if the event type is not supported. So, I put return in the else condition (line 257) above such that the new s3 code wouldn't be ran.

Then if you want to do it before the callback, or in each of the conditions, you're only adding one line.

I don't feel too strongly about going before. Maybe add a comment before the method invocation that it will execute even if the callback has been invoked, since that caused some confusion in the PR review.